### PR TITLE
add support for blockcommit relative path case

### PR DESCRIPTION
**Pull request by nanli1**
add qemu-img rebase and do commit with virsh opt
Signed-off-by: nanli <nanli@redhat.com>

Note:
fedora_35 AVOCADO_SRC: SETUP:setup.py develop --user VT_TYPE:qemu error not related my code ,
Please help review and ignore this error

### CODE REVIEW

**Comment by nanli1** [comment]
@chunfuwen Hi, chunfu ,Could you also please review this pr ,because this function is also sourced from you ,Thank you

File: virttest/utils_libvirt/libvirt_disk.py [hunk 1]
```diff
@@ -271,7 +271,7 @@ def create_reuse_external_snapshots(vm, pre_set_root_dir=None, skip_first_one=Fa
 
 
 def make_syslink_path_backing_files(pre_set_root_dir, volume_path_list, origin_image_format="qcow2",
-                                    created_image_format="qcow2", syslink_back_chain_lenghth=4):
+                                    created_image_format="qcow2", syslink_back_chain_lenghth=4, qemu_rebase=False):
```
Comment (Yingshun) [inline virttest/utils_libvirt/libvirt_disk.py:274]: Please update docstring for 'qemu_rebase'. Thanks!

File: virttest/utils_libvirt/libvirt_disk.py [hunk 2]
```diff
@@ -300,11 +300,13 @@ def make_syslink_path_backing_files(pre_set_root_dir, volume_path_list, origin_i
...
     backing_file_path = _execute_create_backend_file(backing_files_dict, pre_set_root_dir,
-                                                     origin_image_format, created_image_format)
+                                                     origin_image_format, created_image_format, qemu_rebase=qemu_rebase)
     return os.path.join(backing_file_path, "%s.img" % syslink_folder_list[-1]), list(backing_files_dict.values())
 
 
-def _execute_create_backend_file(backing_files_dict, pre_set_root_dir, origin_image_format, created_image_format):
+def _execute_create_backend_file(backing_files_dict, pre_set_root_dir,
+                                 origin_image_format, created_image_format,
+                                 qemu_rebase=False):
```
Comment (Yingshun) [inline virttest/utils_libvirt/libvirt_disk.py:309]: Same as above.

File: virttest/utils_libvirt/libvirt_disk.py [hunk 3]
```diff
@@ -318,14 +320,19 @@ def _execute_create_backend_file(backing_files_dict, pre_set_root_dir, origin_im
...
+            cmd = ("cd %s && qemu-img create -f %s -o backing_file=%s,backing_fmt=%s %s.img" % (
+                backing_file_path, created_image_format, value, disk_format, key))
+
         process.run(cmd, shell=True, ignore_status=False)
         disk_format = created_image_format
     return backing_file_path
 
 
-def do_blockcommit_repeatedly(vm, device_target, options, repeated_counts):
+def do_blockcommit_repeatedly(vm, device_target, options, repeated_counts, **dargs):
```
Comment (Yingshun) [inline virttest/utils_libvirt/libvirt_disk.py:335]: Please update docstring.

File: virttest/utils_libvirt/libvirt_disk.py [hunk 4]
```diff
@@ -318,14 +320,19 @@ def _execute_create_backend_file(backing_files_dict, pre_set_root_dir, origin_im
     disk_format = origin_image_format
     for key, value in list(backing_files_dict.items()):
         backing_file_path = os.path.join(root_dir, key)
-        cmd = ("cd %s && qemu-img create -f %s -o backing_file=%s,backing_fmt=%s %s.img"
-               % (backing_file_path, created_image_format, value, disk_format, key))
+        if qemu_rebase:
+            cmd = ("cd %s && qemu-img rebase -f %s -u -b %s  -F %s %s"
+                   % (backing_file_path, created_image_format, value, disk_format, key+".img"))
```
Comment (dzhengfy) [inline virttest/utils_libvirt/libvirt_disk.py:325]: `key + ".img"`

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: virttest/utils_libvirt/libvirt_disk.py (hunk 1 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_disk.py:274]: Thanks yingshun, Done

File: virttest/utils_libvirt/libvirt_disk.py (hunk 2 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_disk.py:310]: Thanks yingshun, Done

File: virttest/utils_libvirt/libvirt_disk.py (hunk 3 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_disk.py:337]: Thanks yingshun, Done :D

File: virttest/utils_libvirt/libvirt_disk.py (hunk 4 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_disk.py]: Thanks dzheng, Done

**Comment by nanli1** [comment]
@dzhengfy @Yingshun Hi dzheng, yingshun. Please help re-review

**Review by dzhengfy — APPROVED** [review]
LGTM

**Review by Yingshun — APPROVED** [review]

**Pull request merged**